float64[] position
bool isReady
